### PR TITLE
INTMDB-414: project update. Changes includes adding Parameters according latest API documentation and atlas-go-client. New functionality for asssociating apikeys and teams in project. Documentation changes will be taken in Separate PR

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: cfn-resources/project/cmd/resource/resource.go
Comment (ParthasarathyV) [inline cfn-resources/project/cmd/resource/resource.go:62]: Code to add API keys and Teams during Create

**Review by ParthasarathyVaradhan — APPROVED** [review]
Addressed
keys and teams as part of Project Creation
cfn test is passing

**Comment by Zuhairahmed** [comment]
Can you double check if we need any updates to Project Example (https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples/project)? Also please add link to Project Example in documentation so users know to check there.

File: cfn-resources/project/docs/README.md
Comment (Zuhairahmed) [inline cfn-resources/project/docs/README.md:14]: For MongoDB::Atlas::Project description at top we should copy the Documentation definition as close as possible to prevent confusion. For Project it is  "Returns, adds, and edits collections of clusters and users in MongoDB Cloud." https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Projects

File: cfn-resources/project/docs/projectapikey.md
Comment (Zuhairahmed) [inline cfn-resources/project/docs/projectapikey.md:1]: minor point, we call this "ProjectApiKeys" in README (upper case "P"). if there is not a strong reason to deviate, can we follow same format here too on each of the dedicated pages? also check "ProjectSettings" and "ProjectTeam"

File: cfn-resources/project/docs/README.md
Comment (Zuhairahmed) [inline cfn-resources/project/docs/README.md:102]:
```suggestion
#### ProjectTeam
```

**Comment by Zuhairahmed** [comment]
How did we decide to create dedicated pages for some properties and not others? For example we have specific pages that go into more detail only for: ProjectApiKey, ProjectSettings, ProjectTeam

**Comment by Zuhairahmed** [comment]
Do we still need MongoDB::Atlas::Project apiKeyDefinition? If not delete it. https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/cfn-resources/project/docs/apikeydefinition.md

**Review by Zuhairahmed — CHANGES_REQUESTED** [review]
overall great start, left a few comments above. ping me if you need to chat on anything

File: cfn-resources/go.mod
Comment (govardhanpagidi) [reply cfn-resources/go.mod:13]: Actually this is bumped and checked in the #113. I ll pull  once we merge it.

File: cfn-resources/project/docs/projectapikey.md
Comment (govardhanpagidi) [reply cfn-resources/project/docs/projectapikey.md:1]: This is the definition of ProjectApiKey. Project has the Property as ProjectApiKeys : with  [] ProjectApiKey.

File: cfn-resources/project/docs/README.md
Comment (govardhanpagidi) [reply cfn-resources/project/docs/README.md:102]: We can add multiple Teams to the project, so it is considered to be an array name. ProjectTeams : [] ProjectTeam

File: cfn-resources/project/docs/README.md
Comment (govardhanpagidi) [reply cfn-resources/project/docs/README.md:14]: Addressed in the latest commit.

Comment (govardhanpagidi) [reply cfn-resources/project/docs/README.md:14]: It doesn't sound correct to me, still added as suggested.

**Comment by govardhanpagidi** [comment]
Cfn template example has been updated for Project.

**Comment by matt-condon** [comment]
Could you please add some detail in the PR description of the desired changes (for visibility of reviewers)?

File: cfn-resources/project/docs/README.md
Comment (Zuhairahmed) [reply cfn-resources/project/docs/README.md:14]: Thanks. you're welcome to open another doc PR to change definition in our OpenAPI docs :) this likely goes back to use case of almost all customers leveraging Atlas to create clusters and you can't create a cluster without first creating a Project

Comment (themantissa) [reply cfn-resources/project/docs/README.md:14]: It's not a very good definition, agree.  I think what it is trying to express is that a lot of settings that impact clusters and users are at the project level.  So from an organizational perspective this is true.

Comment (Zuhairahmed) [reply cfn-resources/project/docs/README.md:14]: Let's revisit with the dedicated Documentation PR for MongoDB::Atlas::Project as discussed. Thanks @govardhanpagidi !

**Review by Zuhairahmed — APPROVED** [review]
Let's revisit with the dedicated Documentation PR for MongoDB::Atlas::Project as discussed. Thanks @govardhanpagidi ! LGTM, pending APIx team review

**Review by matt-condon — APPROVED** [review]
LGTM once conflicts are addressed

**Comment by govardhanpagidi** [comment]
@matt-condon , I resolved the conflicts.

**Pull request merged**
